File: Data Structures/trie.py
class TrieNode:
    def __init__(self,data):
        self.mapping = {}
        self.endofword = False
        self.data = data
    
class Trie:
    def __init__(self):
        self.root=TrieNode(0)
        self.count = 1
    
    def addWord(self,s):
        p = self.root
        for i in range(len(s)):
            if s[i] not in p.mapping:
                p.mapping[s[i]] = TrieNode(self.count)
                self.count+=1
            p = p.mapping[s[i]]
        p.endofword = True
    


        

File: Data Structures/trie.java
import java.util.*;

class TrieNode{
    boolean endOfWord;
    HashMap<Character,TrieNode> mapping;

    TrieNode(){
        endOfWord = false;
        mapping = new HashMap<>();
    }
}

class Trie{
    TrieNode root;

    Trie(){
        root = new TrieNode();
    }

    void addWord(String s){
        TrieNode p = root;
        for(int i=0;i<s.length();i++){
            if(!p.mapping.containsKey(s.charAt(i)))
                p.mapping.put(s.charAt(i), new TrieNode());    
            p = p.mapping.get(s.charAt(i));
        }
        p.endOfWord = true;
    }

    boolean searchWord(String s){
        TrieNode p = root;
        for(int i=0;i<s.length();i++){
            if(!p.mapping.containsKey(s.charAt(i)))
                return false;
            p = p.mapping.get(s.charAt(i));
        }
        return p.endOfWord;
    }
}